src/app/shared/item-card/item-card.component.ts

```diff
@@ -12,10 +18,31 @@ export class ItemCardComponent implements OnInit {
   @Input() item: Item;
 
   private mobileWidth: number;
+  private ready: boolean = false;
+  private basePath:string = '/uploads';
 
-  constructor(private windowService: WindowService) {}
+  constructor(
+    private windowService: WindowService,
+    private dataService: DataService,
+    private db: AngularFireDatabase) { }
 
   ngOnInit() {
     this.mobileWidth = MOBILE_WIDTH;
+
+    var id = String(localStorage.getItem('item' + this.item.id));
```

Comment:
This code block is causing an error when I open up the `homepage.component.ts`

This is the error I am getting in the console:

```
HomepageComponent.html:3 ERROR Error: Reference.child failed: First argument was an invalid path = "/uploads/https://www.obaku.com/content/collection/V213GUCURZ.jpg". Paths must be non-empty strings and can't contain ".", "#", "$", "[", or "]"
    at Object.exports.validatePathString (validation.js:282)
    at Object.exports.validateRootPathString (validation.js:293)
    at Reference.child (Reference.js:72)
    at Database.ref (Database.js:58)
    at ItemCardComponent.ngOnInit (item-card.component.ts:34)
    at checkAndUpdateDirectiveInline (core.js:12352)
    at checkAndUpdateNodeInline (core.js:13876)
    at checkAndUpdateNode (core.js:13819)
    at debugCheckAndUpdateNode (core.js:14712)
    at debugCheckDirectivesFn (core.js:14653)
```